### PR TITLE
fix(DB/Spells): Spell Axe Flurry should target only one player.

### LINKED ISSUE #12525: [ZG] Gurubashi Axe Thrower - Axe Flurry has the wrong behaviour
### Current Behaviour

Axe Flurry channeling causes Axe Flurry (24020) to hit everyone nearby at once.

### Expected Blizzlike Behaviour

Every second, while channeling, the channeling should trigger Axe Flurry (24020) cast at a random target (probably either the current target and/or anyone in the creature's threat table), to a total of 10 times.
Every second, while channeling, the channeling should trigger Axe Flurry (24071) at all nearby targets, to a total of 10 times.

### Source

sniffs from TBCC

### Steps to reproduce the problem

1. `.go c id 11350` with 2 characters.
2. see if only one gets hit by Axe Flurry (24020) at a time.

### Extra Notes

The channeling is triggered by spell 24018, I think.

### AC rev. hash/commit

https://github.com/azerothcore/azerothcore-wotlk/commit/a7b27238238fd5683c8da25b0d8c13e3b5cf4a1b

### Operating system

Windows 10

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by UltraNix**
Fixes #12525

## Issues Addressed:

- Closes #12525

## Tests Performed:

- Not tested.

## How to Test the Changes:

`.go c id 11350 with 2 characters.`
see if only one gets hit by Axe Flurry (24020) at a time.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
Didn't work for me, both got stunned either way.

**Comment by UltraNix** [comment]
@Gultask Please, retest it.

**Comment by Gultask** [comment]
Works great! 👍

File: src/server/scripts/EasternKingdoms/ZulGurub/zulgurub.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    SPELL_AXE_FLURRY_STUN   = 24020
+};
+
+class spell_axe_flurry : public SpellScript
+{
+    PrepareSpellScript(spell_axe_flurry);
+
+    void FilterTargets(std::list<WorldObject*>& targets)
+    {
+        targets.remove_if(Acore::UnitAuraCheck(true, SPELL_AXE_FLURRY_STUN));
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/ZulGurub/zulgurub.cpp:32]: build complaining

File: src/server/scripts/EasternKingdoms/ZulGurub/zulgurub.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        Acore::Containers::RandomResize(targets, 1);
+    }
+
+    void HandelEffect(SpellEffIndex /*effIndex*/)
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/ZulGurub/zulgurub.cpp:36]: Handel -> Handle

File: src/server/scripts/EasternKingdoms/ZulGurub/zulgurub.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            {
+                caster->CastSpell(target, SPELL_AXE_FLURRY_STUN, true);
+            }
+        }
+    }
+
+    void Register() override
+    {
+        OnObjectAreaTargetSelect += SpellObjectAreaTargetSelectFn(spell_axe_flurry::FilterTargets, EFFECT_0, TARGET_UNIT_SRC_AREA_ENEMY);
+        OnEffectHitTarget += SpellEffectFn(spell_axe_flurry::HandelEffect, EFFECT_0, SPELL_EFFECT_DUMMY);
```
Comment (Nefertumm) [inline src/server/scripts/EasternKingdoms/ZulGurub/zulgurub.cpp:50]: Same as above

**Review by Nefertumm — DISMISSED** [review]

**Comment by UltraNix** [comment]
@Nefertumm Done.

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**